test/lib/deployment/test-deployment.js

```diff
@@ -348,6 +352,34 @@ async function testDeployment(fixturePath, opts) {
   const uploadNowJson = nowJson.uploadNowJson;
   delete nowJson.uploadNowJson;
 
+  // Set `projectSettings.nodeVersion` based on the "engines.node" field of
```

Comment:
This is probably the most important change in this PR. We mimic the logic in CLI which reads the `engines.node` field from package.json and sets that as `projectSettings.nodeVersion` in the create deployment POST body. This covers most of the Node.js-related test fixtures.